reprounzip-singularity/Singularity_unpacker.py

```diff
@@ -0,0 +1,236 @@
+    upload_download_cmd += "download()\n{\n"
+    upload_download_cmd += "\tcp {0}/".format(workingdir) + "\"$1\" " + MOUNT_DIR + "\n}\n\n"
+    upload_download_cmd += "if [ \"$#\" -gt 0 ]\nthen \n \tcase \"$1\" in \n\t\t\"upload\"|\"download\")\n\t\t\t\"$1\" \"$2\" \n\t\t;;\n\t\t*)\n\t\techo \"Invalid arguments!!\" >&2\n\t\texit 1\n\t\t;;\n\tesac\nelse\n\trun\nfi"
+    with open(os.path.join(SINGULARITY_DIR, "runscript"), 'w') as f:
+	f.write(run_cmd)
+        f.write(upload_download_cmd)
+
+def make_main_app_base_script(app_name):
+    print("rached 94")
+    cmd =  "SCIF_APPDATA_{0}=/scif/data/{1}\n".format(app_name,app_name)
+    cmd += "SCIF_APPMETA_{0}=/scif/apps/{1}/scif\n".format(app_name,app_name)
+    cmd += "SCIF_APPROOT_{0}=/scif/apps/{1}\n".format(app_name,app_name)
+    cmd += "SCIF_APPBIN_{0}=/scif/apps/{1}/bin\n".format(app_name,app_name)
+    cmd += "SCIF_APPLIB_{0}=/scif/apps/{1}/lib\n".format(app_name,app_name)
+    cmd += "export SCIF_APPDATA_{0} SCIF_APPROOT_{1} SCIF_APPMETA_{2} SCIF_APPBIN_{3} SCIF_APPLIB_{4}\n".format(app_name,app_name,app_name,app_name,app_name)
+    cmd += "SCIF_APPENV_{0}=/scif/apps/{1}/scif/env/90-environment.sh\n".format(app_name,app_name)
+    cmd += "export SCIF_APPENV_{0}\n".format(app_name)
+    cmd += "SCIF_APPLABELS_{0}=/scif/apps/{1}/scif/labels.json\n".format(app_name,app_name)
+    cmd += "export SCIF_APPLABELS_{0}\n".format(app_name)
+    cmd += "SCIF_APPRUN_{0}=/scif/apps/{1}/scif/runscript\n".format(app_name,app_name)
+    cmd += "export SCIF_APPRUN_{0}\n".format(app_name)
+    with open(os.path.join(ENV_DIR, "94-appsbase.sh"), 'a') as f:
+        f.write(cmd)
+
+
+def make_app_specific_base_script(app_name):
+    cmd = "SCIF_APPNAME={}\n".format(app_name)
+    cmd += "SCIF_APPROOT=\"/scif/apps/{}\"\n".format(app_name)
+    cmd += "SCIF_APPMETA=\"/scif/apps/{}/scif\"\n".format(app_name)
+    cmd += "SCIF_DATA=\"/scif/data\"\n"
+    cmd += "SCIF_APPDATA=\"/scif/data/{}\"\n".format(app_name)
+    cmd += "SCIF_APPINPUT=\"/scif/data/{}/input\"\n".format(app_name)
+    cmd += "SCIF_APPOUTPUT=\"/scif/data/{}/output\"\n".format(app_name)
+    cmd += "export SCIF_APPDATA SCIF_APPNAME SCIF_APPROOT SCIF_APPMETA SCIF_APPINPUT SCIF_APPOUTPUT SCIF_DATA\n"
+    with open(os.path.join(APP_ENV_DIR, "01-base.sh"), 'w') as f:
+        f.write(cmd)
+
+# Check if bin is present in the tar if not add bin and sh
+def copy_busybox(tar):
+    if not "bin" in tar.getnames():
+        print("bin is absent")
+        tar.add("../../bin",arcname="bin")
+    elif not "sh" in tar.getnames():
+        print("bin but no sh")
+        tar.add("../../bin/sh", arcname='bin/sh')
+
+
+def deleteContent(fName):
+    with open(fName, "w"):
+        pass
+
+def add_singularity_folder(tar):
+    # check if the runs are multiple or single
+    my_dict = yaml.load(open(source_config))
+    runs = my_dict['runs']
+    copy_busybox(tar)
+    deleteContent(os.path.join(ENV_DIR, "94-appsbase.sh"))
+    if len(runs) > 1:
```

Comment:
Can we just always create the apps? I don't see a problem with there only being one.